### PR TITLE
fix/97 - simulation issues

### PR DESCRIPTION
**Pull request by corymosiman12**
Initial work started for: https://github.com/BuildingSync/BuildingSync-gem/issues/97

- cleaned up redundant parameters passing, `method(element, ns)` -> `method` by initializing superclasses for `LocationElement`, `SpatialElement`, etc.
- removed awkward weather dir lookup
- strict requirements on single facility

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,31 +31,50 @@ jobs:
       - $SH "bundle exec rspec spec/tests/model_articulation_test/facility_spec.rb"
       - $SH "bundle exec rspec spec/tests/model_articulation_test/hvac_system_spec.rb"
       - $SH "bundle exec rspec spec/tests/model_articulation_test/loads_system_spec.rb"
+      - $SH "bundle exec rspec spec/tests/model_articulation_test/lighting_system_type_spec.rb"
       - $SH "bundle exec rspec spec/tests/model_articulation_test/service_hot_water_system_spec.rb"
       - $SH "bundle exec rspec spec/tests/model_articulation_test/site_spec.rb"
       # - $SH "bundle exec rspec spec/tests/model_articulation_test/weather_file_download_spec.rb"  // fails sometimes due to connection issues, so we exclude it here
       name: "Model Articulation Tests"
+    - script:
+        - travis_wait 40 $SH "bundle exec rspec spec/tests/model_articulation_test/hospital_occupancy_type_spec.rb"
```
Comment (nllong) [inline .travis.yml:40]: We are going to run out of travis credits pretty soon. Leave this as-is, but we will need to move over to github actions.

File: Rakefile [hunk 2]
```diff
@@ -47,4 +49,29 @@ require 'openstudio/model_articulation'
 rake_task = OpenStudio::Extension::RakeTask.new
 rake_task.set_extension_class(OpenStudio::ModelArticulation::Extension)
 
+desc 'Convert tabs to spaces'
+task :remove_tabs do
```
Comment (nllong) [inline Rakefile:53]: yes!

File: lib/buildingsync/constants.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+SCHEMA_2_0_URL = 'https://raw.githubusercontent.com/BuildingSync/schema/v2.0/BuildingSync.xsd'
```
Comment (nllong) [inline lib/buildingsync/constants.rb:40]: thanks for making these obvious. Is there documentation (in the README) saying to update these when upgrading the version of BuidngSync?

File: lib/buildingsync/helpers/helper.rb [hunk 4]
```diff
@@ -36,31 +38,136 @@
...
+    # @return [String] if text value exists
+    # @return [nil] if text value doesnt exist or element is complex (has children)
+    def help_get_text_value(xml_element)
+      if xml_element && !xml_element.text.nil?
         return xml_element.text
       end
       return nil
     end
 
+    def help_get_text_value_as_float(xml_element)
```
Comment (nllong) [inline lib/buildingsync/helpers/helper.rb:68]: nice helpers!

**Review by nllong — APPROVED** [review]
holy cow you did a bunch of work on this! Good job!

I started going through the details. In general your coding looks good and you have followed good conventions. I am good with this if the tests pass.

Again, awesome job getting this all working!

File: .travis.yml (hunk 1 above)
Comment (corymosiman12) [reply .travis.yml:40]: we are already out...

Comment (nllong) [reply .travis.yml:40]: travis definitely fits in 2020

**Comment by corymosiman12** [comment]
Still some things outstanding:
- #100 Refactor Systems 
- L100 simulations no longer working due to above.  Just needs better logic.
- #101 DEER vs. CBES
- #102 Improve HVACSystem Selection
- #103 SmallHotel

## Tests failing:
### model_articulation_test/*
- 8 total
- 4 tests from `facility_spec.rb`.  Minor fixes needed.
- 2 tests from `weather_file_download_spec.rb` - this fails sometimes due to connection issues.  noted in `.travis.yml`
- 1 for SmallHotel - see previous
- 1 from `site_spec.rb` - OSM generated not same as previous...need to look into.

**Pull request merged**